contents/ruby/Rails.md

```diff
@@ -0,0 +1,170 @@
+
+```ruby
+resources :users
+```
+
+Now, you can check all the routes that you have in your application by running `rails routes` in the command line. This should include:
+
+```
+                   Prefix Verb   URI Pattern                                                                              Controller#Action
+                    users GET    /users(.:format)                                                                         users#index
+                          POST   /users(.:format)                                                                         users#create
+                 new_user GET    /users/new(.:format)                                                                     users#new
+                edit_user GET    /users/:id/edit(.:format)                                                                users#edit
+                     user GET    /users/:id(.:format)                                                                     users#show
+                          PATCH  /users/:id(.:format)                                                                     users#update
+                          PUT    /users/:id(.:format)                                                                     users#update
+                          DELETE /users/:id(.:format)                                                                     users#destroy
+```
+
+As can be seen, the common routes that you would need for the resources have already been mapped to their respective URI's. This is equivalent to manually declaring the 7 resourceful routes for the `user` resource. By convention, Rails will then look for a controller called `users_controller.rb` and invoke the name of the function mapped to the particular URI and Verb. More information about Rails routes and how they work can be found [here](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default).
+
+#### Controller
+In the standard Rails convention, once inside the application directory, if you navigate to `app/controllers/`, you should be able to see all the controllers in your project. Rails will look for the controller with the `<name>_controller.rb`, where `<name>` would be replaced by the result under the `Controller` heading in the list of routes above, in this case, "users". It will then look for the function name under the `Action` heading mapped to the URI function inside the Controller, and then call it. This is where the bulk of the business logic of your application would be. So if the person makes a GET request to `/users`, then Rails will open `users_controller.rb`, and call its `index` method (refer to the code below).
```

Comment:
"users" -> `users`

Reply:
> This is where the bulk of the business logic of your application would be.

Based on Rails best practices, the bulk of the business logic should not live in the controller. This sentence also feels like a mini sidetrack. I think it'll be better to remove it so there's no break when describing the control flow (`...and then call it.` immediately followed by `So if the person makes a GET request...`)

Reply:
And maybe the last sentence should instead read as:

For example, if someone makes a GET request to `/users`, Rails will then respond by calling the `users#index` controller action (refer to the code below).

"So if the person" is super conversational... 😛